packages/lib/src/vis/tiles/TiledHeatmap.tsx

```diff
@@ -0,0 +1,76 @@
+import { clamp, range } from 'lodash';
+import { Color } from 'three';
+
+import { getInterpolator } from '../heatmap/utils';
+import { useFrameRendering } from '../hooks';
+import { useAxisSystemContext } from '../shared/AxisSystemContext';
+import TiledHeatmapLayer from './TiledHeatmapLayer';
+import type { TilesApi } from './api';
+import { useDataPerPixel } from './hooks';
+import type { ColorMapAttributes } from './models';
+
+interface Props extends ColorMapAttributes {
+  api: TilesApi;
+  progressive?: boolean;
+  qualityFactor?: number;
+}
+
+function TiledHeatmap(props: Props) {
+  const {
+    api,
+    progressive = true,
+    qualityFactor = 1, // 0: Lower quality, less fetch; 1: Best quality
+    ...colorMapAttrs
+  } = props;
+
+  const { visSize } = useAxisSystemContext();
+  const { xDataPerPixel, yDataPerPixel } = useDataPerPixel();
+
+  const nLevels = api.levelsSize.length;
+  const [level0Size] = api.levelsSize;
+
+  const quality = 1 - clamp(qualityFactor, 0, 1);
+  const levelOfDetail = Math.min(
+    Math.floor(Math.log2(Math.max(1, xDataPerPixel, yDataPerPixel)) + quality),
+    Math.max(nLevels - 1, 0)
+  );
+
+  const lods = progressive
+    ? range(levelOfDetail, nLevels).reverse()
+    : [levelOfDetail];
+
+  const { colorMap, invertColorMap = false } = colorMapAttrs;
+
+  useFrameRendering();
+
+  // Work in coordinates of level 0 array
+  return (
+    <group
+      position={[-visSize.width / 2, -visSize.height / 2, 0]}
+      scale={[
+        visSize.width / level0Size.width,
+        visSize.height / level0Size.height,
+        0,
+      ]}
+    >
+      <mesh position={[level0Size.width / 2, level0Size.height / 2, -0.1]}>
+        <planeGeometry args={[level0Size.width, level0Size.height]} />
+        <meshBasicMaterial
+          color={new Color(getInterpolator(colorMap, invertColorMap)(0))}
```

Comment:
Correct me if I am wrong, but I don't think the `Color` constructor is needed ?

Reply:
typescript complains if `new` is removed

Reply:
Sorry I wasn't clear: I mean that doing `color={getInterpolator(colorMap, invertColorMap)(0)}` might be enough (no need to wrap it in a `Color` object)

Reply:
OK, done!